metals/src/main/scala/scala/meta/metals/providers/CompletionProvider.scala

```diff
@@ -19,16 +19,26 @@ object CompletionProvider extends LazyLogging {
   ): CompletionList = {
     import compiler._
 
+    val functionTraitRegex = """^scala\.Function([0-9]|1[0-9]|2[0-2])$""".r
+    def isFunction(symbol: Symbol): Boolean = {
+      val typeSymbolFullName = symbol.info.finalResultType.typeSymbol.fullName
+      functionTraitRegex.findFirstIn(typeSymbolFullName).isDefined
+    }
```

Comment:
You can query the compiler directly for this piece of information with

```scala
compiler.definitions.isFunctionSymbol(
  symbol.info.finalResultType.typeSymbol
)
```

(thanks @olafurpg for the tip)

Reply:
I spent way too long trying to figure something like that out before settling on the ugly regex solution... thanks